File: app/app/src/main/java/com/hobbymeetingapp/app/navigation_bar/search_events/EventCardsAdapter.java
package com.hobbymeetingapp.app.navigation_bar.search_events;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.hobbymeetingapp.app.R;

public class EventCardsAdapter extends RecyclerView.Adapter<EventCardsAdapter.MyViewHolder> {
    private String[] mDataset;

    static class MyViewHolder extends RecyclerView.ViewHolder {
        TextView textView;
        MyViewHolder(TextView v) {
            super(v);
            textView = v;
        }
    }

    public EventCardsAdapter(String[] myDataset) {
        mDataset = myDataset;
    }


    @NonNull
    @Override
    public EventCardsAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent,
                                                             int viewType) {

        TextView v = (TextView) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.member_item, parent, false);
        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        holder.textView.setText(mDataset[position]);

    }

    @Override
    public int getItemCount() {
        return mDataset.length;
    }
}

File: app/app/src/main/java/com/cabal/app/navigation_bar/search_events/EventCardsAdapter.kt
package com.cabal.app.navigation_bar.search_events

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.cabal.app.R
import com.cabal.app.database.entities.User
import java.util.stream.Collectors

class EventCardsAdapter(dataset: List<User>) : RecyclerView.Adapter<EventCardsAdapter.MyViewHolder>() {

    private val data: List<String?> = dataset.stream().map { it.nick }.collect(Collectors.toList())


    class MyViewHolder(val view: TextView) : RecyclerView.ViewHolder(view)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder =
            MyViewHolder(LayoutInflater
                    .from(parent.context)
                    .inflate(R.layout.member_item, parent, false) as TextView)

    override fun getItemCount(): Int = data.size

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) =
            let {holder.view.text = data[position] }
}